TestErrorsTemporary fails with "can't assign requested address"

### Your environment.
  - Version: 38e39e4
  - Browser: n/a

### What did you do?
Ran unit tests on my mac:
```
% sw_vers
ProductName:		macOS
ProductVersion:		14.2.1
BuildVersion:		23C71
```

### What did you expect?
All tests under pion/dtls to pass.

### What happened?
```
=== RUN   TestErrorsTemporary
    /Users/ytakeda/Projects/pion/dtls/errors_errno_test.go:26: Unexpected error: dial udp 127.0.0.1:0: connect: can't assign requested address
--- FAIL: TestErrorsTemporary (0.00s)
FAIL
FAIL	github.com/pion/dtls/v2	0.203s
```

--- a/errors_errno_test.go
+++ b/errors_errno_test.go
@@ -16,18 +16,32 @@ import (
 )
 
 func TestErrorsTemporary(t *testing.T) {
-	addrListen, errListen := net.ResolveUDPAddr("udp", "localhost:0")
-	if errListen != nil {
-		t.Fatalf("Unexpected error: %v", errListen)
+	// Allocate a UDP port no one is listening on.
+	addrListen, err := net.ResolveUDPAddr("udp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Unexpected failure to resolve: %v", err)
 	}
+	listener, err := net.ListenUDP("udp", addrListen)
+	if err != nil {
+		t.Fatalf("Unexpected failure to listen: %v", err)
+	}
+	raddr, ok := listener.LocalAddr().(*net.UDPAddr)
+	if !ok {
+		t.Fatal("Unexpedted type assertion error")
+	}
+	err = listener.Close()
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
 	// Server is not listening.
-	conn, errDial := net.DialUDP("udp", nil, addrListen)
+	conn, errDial := net.DialUDP("udp", nil, raddr)
 	if errDial != nil {
 		t.Fatalf("Unexpected error: %v", errDial)
 	}
 
 	_, _ = conn.Write([]byte{0x00}) // trigger
-	_, err := conn.Read(make([]byte, 10))
+	_, err = conn.Read(make([]byte, 10))
 	_ = conn.Close()
 
 	if err == nil {

